recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,478 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import re
+
+
+class CPythonConan(ConanFile):
+    name = "cpython"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.python.org"
+    description = "Python is a programming language that lets you work quickly and integrate systems more effectively."
+    topics = ("conan", "python", "cpython", "language", "script")
+    license = ("Python-2.0",)
+    exports_sources = "patches/**"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "optimizations": [True, False],
```

Comment:
Thanks for the hint.
The options lto, docstrings and pymalloc have the same problem.
For now, I'll throw an error if these options are enabled on Visual Studio.

Reply:
So if I understand it correctly, first build the projects with `PGInstrument` build_type, then `PGUpdate`, and only then build the final binaries.

This doesn't work currently, because the `msbuild` conan generator only provides the dependencies for the common build_types (Release, Debug, ...).

Reply:
I forgot that I removed lto, docstrings and pymalloc. Doh...

Reply:
Generally the msvc build is much less configurable, so I think it is fine to remove most of the options. I would keep the optimization option since the official binaries are built with pgo.

Reply:
> So if I understand it correctly, first build the projects with `PGInstrument` build_type, then `PGUpdate`, and only then build the final binaries.

Not quite. You use the `PGInstrument` build target, then you run the python test suite with the pgo flag on the resulting binary, and then you run `PGUpdate`. After the `PGUpdate`, you already have your final binaries (i.e. you only compile twice -- once with `PGInstrument` and once with `PGUpdate`).

This is the same basic process that is used when build with optimizations enabled with autotools, it just automates it all in the makefile (and to be fair, the build.bat automates this as well).

Reply:
I'll look into `build.bat` again.
Now that the visual studio projects have been patches to not use vendored dependencies, maybe the script doesn't need much rework.

Reply:
The batch toolchain of python is a series of python scripts for finding msbuild, msvc, ...
Indeed, it also handles pgo, building `PGInstrument` and `PGUpdate`, and running a generated `python.bat` in between.

But I think the current recipe is very easy to understand and adaptable to pgo.
I'm going to keep it 😄

Reply:
I don't disagree 😃.